docs/preprocessors/typescript.md

```diff
@@ -0,0 +1,169 @@
+
+## Getting it to work for your build
+
+For the editor, this is already enough - nothing more to do. But you also need to enhance your build config. Using Rollup, this will work with Svelte and TypeScript as long as you enable `svelte-preprocess` and `@rollup/plugin-typescript`:
+
+-   Install these packages `npm i -D svelte-preprocess typescript tslib @rollup/plugin-typescript`
+-   Add these lines to `rollup.config.js`:
+
+```js
+// ...
+import sveltePreprocess from 'svelte-preprocess';
+import typescript from '@rollup/plugin-typescript';
+
+// ...
+  plugins: [
+    svelte({
+      // ...
+      preprocess: sveltePreprocess(), // <--
+    }),
+
+    // ...
+    commonjs(),
+    typescript(), // <-- added below commonjs
+    // ...
+```
+
+-   Add a `tsconfig.json` with these lines:
+
+```json
+{
+    "include": ["src/**/*"],
+    "exclude": ["node_modules/*", "__sapper__/*", "public/*"],
+    "compilerOptions": {
+        "moduleResolution": "node",
+        "sourceMap": true,
+        "target": "es2017",
+        "types": ["svelte"]
+    }
+}
+```
+
+And this should work to enable full TypeScript checking in your Svelte files. For further discussion and a clonable template [see this issue](https://github.com/sveltejs/language-tools/issues/161).
+
+## Example configuration for the editor
+
+#### Using [svelte-preprocess](https://github.com/sveltejs/svelte-preprocess)
+
+##### Install
+
+```sh
+npm i -D svelte-preprocess typescript
+```
+
+<details>
+<summary>Yarn</summary>
+
+```sh
+yarn add --dev svelte-preprocess typescript
+```
+
+</details>
+
+##### Set up `svelte.config.js`
+
+```js
+const sveltePreprocess = require('svelte-preprocess');
+
+module.exports = {
+    preprocess: sveltePreprocess(),
+};
+```
+
+##### Restart the svelte language server
+
+You will need to tell svelte-vscode to restart the svelte language server in order to pick up the new configuration.
+
+Hit `ctrl-shift-p` or `cmd-shift-p` on mac, type `svelte restart`, and select `Svelte: Restart Language Server`. Any errors you were seeing should now go away and you're now all set up!
+
+## Troubleshooting / FAQ
+
+### How do I type reactive assignments? / I get an "implicitly has type 'any' error"
+
+The following code may throw an error like `Variable 'show' implicitly has type 'any' in some locations where its type cannot be determined.`, if you have stricter type settings:
+
+```html
+<script lang="typescript">
+    export let data: { someKey: string | null };
+
+    $: show = !!data.someKey; // <-- `show` has type `any`
+</script>
+
+{#if show}hey{/if}
+```
+
+To type the variable, do this:
+
+```ts
+let show: boolean; // <--- added above the reactive assignment
+$: show = !!data.someKey; // <-- `show` now has type `boolean`
+```
+
+The following is INVALID TypeScript, so DON'T do this:
+
+```ts
+$: show = !!data.someKey as boolean; // DON'T DO THIS
+$: show: boolean = !!data.someKey; // DON'T DO THIS
+```
```

Comment:
@dummdidumm I think it is unnecessary to add this, because there's a million ways to do the same thing incorrectly, and there's no point in describing just one that is explicitly marked as and error by IDE.

Reply:
Good point, I'll remove it.